pkg/kubelet/kubelet.go

```diff
@@ -2244,6 +2244,27 @@ func (kl *Kubelet) getPodsToSync() []*v1.Pod {
 	return podsToSync
 }
 
+// Ensure node is registered before syncing static pods.
+func (kl *Kubelet) syncStaticPodsOnNodeRegistration() {
+	nodeCh := make(chan struct{})
+	wait.Until(func() {
+		if _, err := kl.GetNode(); err != nil {
+			return
+		}
+		close(nodeCh)
+	}, time.Second, nodeCh)
+
+	allPods := kl.podManager.GetPods()
+	staticPods := []*v1.Pod{}
+	for _, pod := range allPods {
+		if kubetypes.IsStaticPod(pod) {
+			staticPods = append(staticPods, pod)
+		}
+	}
```

Comment:
nit:
```suggestion
        _, staticPods := splitPodsByStatic(kl.podManager.GetPods())
```

Reply:
alternatively you can use [slices.DeleteFunc](https://pkg.go.dev/slices#DeleteFunc), but that would be more wordy.